services/learn-card-network/brain-service/src/routes/boosts.ts

```diff
@@ -119,6 +119,42 @@ export const boostsRouter = t.router({
             return getBoostUri(boost.id, ctx.domain);
         }),
 
+    getBoost: profileRoute
+        .meta({
+            openapi: {
+                protect: true,
+                method: 'GET',
+                path: '/boost/{uri}',
+                tags: ['Boosts'],
+                summary: 'Get boost',
+                description: 'This endpoint gets metadata about a boost',
+            },
+        })
+        .input(z.object({ uri: z.string() }))
+        .output(BoostValidator.omit({ id: true }).extend({ uri: z.string() }))
+        .query(async ({ ctx, input }) => {
+            const { profile } = ctx.user;
+
+            const { uri } = input;
+
+            const boost = await getBoostByUri(uri);
+
+            if (!boost) throw new TRPCError({ code: 'NOT_FOUND', message: 'Could not find boost' });
+
+            if (!(await isProfileBoostOwner(profile, boost))) {
+                throw new TRPCError({
+                    code: 'UNAUTHORIZED',
+                    message: 'Profile does not own boost',
+                });
+            }
+
```

Comment:
This is a question for @gerardopar - what is the use case for getBoost()?

@gerardopar - do you need this query for a logged-out user to be able to retrieve info about a boost before claiming it?

Or, is it okay that getBoost() both a) requires a profile, and b) requires you to be the boost owner?

Reply:
Hey @Custard7 @TaylorBeeston I primarily need this one for checking the status of the boost (LIVE, DRAFT) via the owner, to determine wether or not to disable editing.